Add .conjugate() method to int type

agude opened the pull request:
The new method, `int.conjugate()`, just returns a clone of the `int`.

However, this method is _also_ used by the `bool` type and in that case returning a clone doesn't match CPython.

CPython does:

```Python
>>> type(True.conjugate())
<class 'int'>
```

A couple options (open to others!):

- Implement `bool.conjugate()` independently
- Cast the return type to `int` in `int.conjugate()`

Happy to implement that to, but would love some feedback.

windelbouwman commented:
Hmm, interesting stuff! I did not know this behavior. Is it weird? Maybe. We could also use `objint::get_value` and create a new integer instead of the clone. That would resemble CPython better.

agude commented:
:+1: 

I've updated the function to use `objint::get_value` and return a `new_int`. It now works for both `int`s and `bool`s. I've added a test to make sure `bool.conjugate()` returns an `int` type.

windelbouwman commented:
Cool!
